speedtest/speedtest.go

```diff
@@ -17,11 +42,21 @@ func WithDoer(doer *http.Client) Option {
 	}
 }
 
+// WithUserAgent adds the passed "User-Agent" header to all requests.
+// To use with a custom Doer, "WithDoer" must be passed before WithUserAgent:
+// `New(WithDoer(myDoer), WithUserAgent(myUserAgent))`
+func WithUserAgent(UserAgent string) Option {
+	return func(s *Speedtest) {
+		s.doer.Transport = newUserAgentTransport(s.doer.Transport, UserAgent)
+	}
+}
+
 // New creates a new speedtest client.
 func New(opts ...Option) *Speedtest {
 	s := &Speedtest{
 		doer: http.DefaultClient,
 	}
+	WithUserAgent(defaultUserAgent)(s)
 
 	for _, opt := range opts {
 		opt(s)
```

Comment:
Hi @rtrox. It seems that `WithDoer` and `WithUserAgent` can not be used at the same time, because `WithDoer` will always be executed after `WithUserAgent` in `New(opts ...Option)`.